### DISCUSSION TITLE
Setup with multiple inverters and Smart Logger 3000A

### DISCUSSION THREAD

**Discussion by Roving-Ronin** [wlcrs/huawei_solar#211] [Setting up the integration]
Hi,

Is there anyway to setup this integration to work with 2 x 6kW M1 inverters (each with 19 x 600W optimisers also) that are cascaded together, with the primary inverter then connected to a Huawei Smart Logger 3000A, that is then connected by ethernet (hard wired / RJ45) to the LAN, from where it then uploads to FusionSolar.

The Smart Logger has modbus tcp access enabled, to allow querying from any device upon the home network.

Logical design is as per this diagram:
![Huawei - Dual M1 6kW Inverter Logical Diagram - No ESS or BackupBox](https://user-images.githubusercontent.com/108674933/209336300-5e4a9089-0b94-42d9-a34d-d7ae434d56c9.jpg)

**Comment by wlcrs** [comment]
The [underlying library](https://gitlab.com/Emilv2/huawei-solar) has no support for the SmartLogger 3000A. I do not have such a device, which makes it way too hard for me to implement support.

I'm prepared to assist anybody that is willing to add support for the SmartLogger to the huawei-solar library.  Drop a message in #93 in that case.

**Reply by Roving-Ronin** [reply]
@wlcrs 
Was just about to start (again) trying to work out modbus registers on the SmartLogger (and ordering a wig for the ensuing hair loss), when I thought i'd try adding the integration again (with latest release updates).

Heres my SmartLogger showing the connected inverters and their COM Port addresses, and Logical Addresses.
![image](https://github.com/wlcrs/huawei_solar/assets/108674933/fb3880bf-00ec-411d-806b-21d9651ccd18)


Given the above using the Network option as the connection method, I entered the SmartLogger 3000A's IP address and port 502, and (not knowing if the DTSU-666H is seen as a master) I then tried Slave ID's entered as:

1,2,11   (Primary Inverter, Secondary Inverter and DTSU-666H meter)
11,1,2   (DTSU-666H meter, Primary Inverter, Secondary Inverter)

these failed, **_however_** when entering 1,2 (Primary, Secondary) and enabling '_Advanced: elevate permissions_' (each inverter as 19 x 600W optimisers attached) it works... albeit without the DTSU-666H consumption meter (this is attached via RS485 to COM3 on the SmartLogger, whilst inverters attach to COM1).

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/0cc13814-eb38-4ff7-be6c-6f8213ab788b)
Note: I believe the 38 inverters and their data appear, just when I re-tested this now.... the sun has gone down and the optimisers are offline. D'Oh. Will check again tomorrow.

The only thing that appears to currently not work is being able to 'find' the DTSU-666H (connected on COM3-11. 
Would I be correct in assuming that the integration only queries COM1 (by default) currently ? Or, would you happen to know if there's anyway to configure the integration telling it to query COM1 and COM3 ?

Thanks in Advance !

**Reply by wlcrs** [reply]
I don't know. I'm not familiar with the SmartLogger 3000A apart from some snippets that I read in the documentation.

**Reply by Roving-Ronin** [reply]
@wlcrs OK, but you wrote the integration, so I thought you;d know.... Does it query only Addresses/Slace ID on COM1 (by default) ?  Or can we define COM ports, not just the Address / SlaveID ?

i.e. So when you enter Slave ID's, do you have it written in the code (using this example below), to query 'IP 10.0.5.11 Port 502 and then query COM1 for ID's 1,2,11' ?

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/3c00c7bf-593e-4e6e-8f03-c38214aa976f)

**Reply by Roving-Ronin** [reply]
@wlcrs would you be aware of the output / format of the query from the above screenshot, from the integration calling the inverter (or datalogger in this case) to check what devices are available, and then add them to HA?

Trying to work out if its calling a specific modbus register ?
Or if the format of the SlaveIDs details to calling COM 1 Address (aka SlaveID) 1... and could this work if it was possible to call COM1 Address 11 (enter Slave ID as "3.1" for instance) ?

**Reply by wlcrs** [reply]
Please stop mentioning me. I do not provide any support whatsoever for installations with a SmartLogger. You are on your own. If you are able to get it working, please add your insights here on or the wiki.

**Reply by Roving-Ronin** [reply]
@wlcrs sorry I thought you were the develop for this integration, hence my asking if you might happen to know how the integration 'called' or sends its query to the devices, and the format of such a command.  Idea being with this information I might then be able to work out how to get it to work and then provide such information back to assist with getting it working for everyone.  :-)

**Reply by djsamlb** [reply]
hi I have the EXACT same setup! did u manage to get it working?

**Reply by Roving-Ronin** [reply]
Yes, rewired the DTSU-666H to the primary inverter, plugged in the sDongle and removed the SmartLogger.

:-(

**Reply by 207Engineer** [reply]
I too, have the same setup. In the SmartLogger3000A modbus map, there doesn't appear to be any inverter-specific commands that I can see. It looks like the Active Power Control mode is also Read Only, but can be set from the Enspire HMI that comes with the system.

**Reply by epper202** [reply]
Are you guys still fighting with it?
I managed to pull alot of data from the smartlogger and can control 4 inverters at the same time but w/o this integration.

**Reply by djsamlb** [reply]
> Are you guys still fighting with it? I managed to pull alot of data from the smartlogger and can control 4 inverters at the same time but w/o this integration.

Hey can u give more details plz

**Reply by epper202** [reply]
> > Are you guys still fighting with it? I managed to pull alot of data from the smartlogger and can control 4 inverters at the same time but w/o this integration.
> 
> 
> 
> Hey can u give more details plz

Via modbus tcp integration in home assistant. 
If you google the Modbus smartlogger definitions pdf you have a lot of info to play with. 
Make sure you do enable the NMS function in the huawei environment for a third party server to act as master.

**Reply by Ovik10** [reply]
@epper202 can you please provide me some tutorial on this? I setted up the connection in configuration.yaml but I dont know how to access the data. Thanks

**Reply by epper202** [reply]
> @epper202 can you please provide me some tutorial on this? I setted up the connection in configuration.yaml but I dont know how to access the data. Thanks
> 
> 

Yes, project is on standby with the server off. I can get the config yaml maybe tomorrow

**Reply by Roving-Ronin** [reply]
@epper202 would be interested in all information you have. One of the things that killed me was how to determine the register ID numbers, as the SL300 doco refers to a formula for how they are magically worked out and that varies from the numbers used for non SL3000 connections.
